app/controllers/admin/tests_controller.rb

```diff
@@ -56,6 +56,9 @@ def find_test
   end
 
   def test_params
-    params.require(:test).permit(:title, :level, :category_id)
+    test_pars = params.require(:test).permit(:title, :level, :category_id, :timer, :duration)
+
+    test_pars[:duration] = test_pars[:duration].to_time.seconds_since_midnight unless test_pars[:duration].empty?
```

Comment:
```suggestion
    test_pars[:duration] = prrepare_duration(test_pars[:duration]) if test_pars[:duration].present?
```
и все вычисления вынести во вспомогательный метод